### PR TITLE
TEP-0075: Add examples with object params and results

### PR DESCRIPTION
**Pull request by chuangw6**
# Changes

- Added a taskrun level example that uses object param and
object result.

- Added a pipeline level example that uses individual variables of an
object param and uses the whole object param.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Add taskrun & pipelinerun examples that use object param and result.
```

### CODE REVIEW

File: examples/v1beta1/pipelineruns/alpha/pipeline-object-param.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: mock-clone-task
+spec:
+  params:
```
Comment (ywluogg) [inline examples/v1beta1/pipelineruns/alpha/pipeline-object-param.yaml:6]: Looking at `examples/v1beta1/pipelineruns/alpha/pipeline-emitting-results.yaml` together, I realize another example is missing is that: a task is using a previous task's results in the params as a whole. Example with object keys: https://github.com/tektoncd/pipeline/blob/main/examples/v1beta1/pipelineruns/task_results_example.yaml and https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#passing-one-tasks-results-into-the-parameters-or-when-expressions-of-another

Comment (chuangw6) [reply examples/v1beta1/pipelineruns/alpha/pipeline-object-param.yaml:6]: This file is the example for object param only. Object result (i.e. using a previous task's object result) example is in the file [/pipeline-object-results.yaml](https://github.com/tektoncd/pipeline/blob/main/examples/v1beta1/pipelineruns/alpha/pipeline-object-results.yaml).

File: examples/v1beta1/taskruns/alpha/object-param-result.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        args: [
+          "echo",
+          "--url=$(params.gitrepo.url)",
+          "--commit=$(params.gitrepo.commit)"
+        ]
+      - name: write-object-result
+        image: bash:latest
+        script: |
+          #!/usr/bin/env bash
+          echo -n "{\"IMAGE_URL\":\"ar.com\", \"IMAGE_DIGEST\":\"sha234\"}" > $(results.object-results.path)
```
[the remark this answers is not in the document]
Comment (chuangw6) [reply examples/v1beta1/taskruns/alpha/object-param-result.yaml:36]: sg!

**Comment by ywluogg** [comment]
#4723

**Pull request merged**